cmd/consul-dataplane/merge.go

```diff
@@ -0,0 +1,269 @@
+package main
+
+import (
+	"strings"
+
+	"github.com/hashicorp/consul-dataplane/pkg/consuldp"
+)
+
+// mergeConfigs takes in a couple of configs(c1&c2) and applies
+// c2 on to c1 based on the following conditions
+//
+// 1. If an attribute of c2 is a string/integer/bool, then it is applied to the
+// same field in c1 if it holds non default value for the field
+// 2. If an attribute of c2 is a slice/map, then it overrides the same field in c1
+func mergeConfigs(c1, c2 *consuldp.Config) {
+	mergeConsulConfigs(c1.Consul, c2.Consul)
+	mergeServiceConfigs(c1.Service, c2.Service)
+	mergeTelemetryConfigs(c1.Telemetry, c2.Telemetry)
+	mergeLoggingConfigs(c1.Logging, c2.Logging)
+	mergeEnvoyConfigs(c1.Envoy, c2.Envoy)
+	mergeXDSServerConfigs(c1.XDSServer, c2.XDSServer)
+	mergeDNSServerConfigs(c1.DNSServer, c2.DNSServer)
+}
```

Comment:
@Ganeshrockz - can we please consider - https://github.com/imdario/mergo, to make code more beautiful.
That library uses reflection to figure out keys dynamically. I think that will be a better approach instead of setting each key value pair.

Reply:
Thanks for the suggestion. I was looking for alternatives to the current approach and mergo seems to be the perfect candidate. Added the same.

@pglass It did not increase the image size a lot.